classes/asserter/resolver.php

```
@@ -50,13 +52,28 @@ public function resolve($asserter)
 		{
 			$class = $this->checkClass($asserter);
 		}
-		else foreach ($this->namespaces as $namespace)
+		else
 		{
-			$class = $this->checkClass($namespace . '\\' . $asserter);
+			if (false === test::isValidIdentifier($asserter))
+			{
+				return null;
```

I would rather throw an `invalidArgument` exception here.